src/services/basic-desc/basic-desc.service.ts

```diff
@@ -13,6 +13,13 @@ export class BasicDesc implements OnInit {
 	@Input()
 	props: any;
 
+	@Input()
+	isOnMap: boolean = false;
+
+	private selfPrimitive: any;
+
```

Comment:
check out `ac-billboard.component`. there is a separation between ac-*-desc and ac-*. 
ac-*-desc are description for the ac-layer how to render the entity.ac-* are for entities for the ac-map layer.
It also gives you a better separation of code , instead of one big component that does 2 different things .
please refactor to use ac-* , maybe need to create abstract class something like `basicMapComponent` .